### DISCUSSION TITLE
Player Restraining

### DISCUSSION THREAD

**Discussion by bundoggity** [EveronLife/EveronLife#114] [Ideas]
I have been working on player restraining and would like some input before diving deep.

Proposal: 

Player interaction option to restrain a player. This would then check for handcuffs or zipties in the attacking players inventory. 

Upon using zipties the ziptie is removed and will be a one use item. 

Handcuffs will have a universal key and can also be lockpicked. The handcuffs are discarded but are returned to the player who unlocks them

After a player is restrained they are set to walking speed. Players will then have the option to drag or search the held player.

I would assume the restricted player can be looted to completion. But would like input on this mechanic.

**Comment by BenRauzi** [comment]
Could potentially also have a 'shackle legs' option or something similar that when both are used completely restricts movement.  (Because people are bound to troll).

Gesture animations for restraints are also quite straightforward now which is awesome.

**Reply by bundoggity** [reply]
I think the leg restraints are a good idea. Originally I was going to limit the players movespeed regardless.  Do you think the player would need a separate item to achieve this? Would seem overkill and should be a default option. 

Also regarding the gesturing,  do you mean "hands up" required to actually restrain a player?

**Reply by BenRauzi** [reply]
What I've used before on servers was that restraint (general hand restraining) limited the movement speed to walking, but leg shackles completely froze you still. (Mainly to stop trolls just walking away constantly).

More was just referring to 'gesture animations' (more an RV term that doesn't really matter anymore, can't remember the proper terminology on Enfusion) where it only applies to the bones the animation is on. (So you can still do other actions, e.g. walking, while doing another animation like being restrained, hands up, etc.)

**Reply by bundoggity** [reply]
> (So you can still do other actions, e.g. walking, while doing another animation like being restrained, hands up, etc.)

Ill take a look into this. Haven't even started with models and animations. My handcuffs are an apple. Thanks for your input!

**Reply by TheOriginalCrazyDuck** [reply]
So just a couple of thoughts:

- In the same vain as handcuffs requiring keys, maybe zipties should require a knife.
- An adjustable restrained movement speed might be ideal for flexibility, for example, some servers may prefer to allow restrained people to run to create situations like [this](https://www.youtube.com/watch?v=-NtFifCJg2k)

**Reply by TL6868** [reply]
I think with a feature like this it is important to remember that we are developing a framework and not a server. With the restraint system, I think handcuffs should be able to be both a single-use and an infinite use item and should provide the option of using a key and not needing to use a key. Also providing that choice with the zip ties.

**Reply by BenRauzi** [reply]
Yes of course, it's good to have config options for whether these settings are enabled or not, and to what degree.

**Comment by Arkensor** [comment]
The most important technical aspects to look into are
1. How to limit the player's ability to run (force slow walk)
2. How to limit him from accessing his own inventory or other people's inventory
3. How can other people "search" somebody that is being restrained? (Or who surrendered for that sake)
4. Are there existing animations we can use? If not can we have custom animations for hands behind the head (surrender)? I am pretty sure we can have a "holding" animation for handcuffs that we force into people's hands. So that might be an option there. It could allow animation blending for when you are handcuffed but still walking or sitting in a car maybe. Not sure.

Once we know what is technically possible we can think about how a balanced gameplay interaction for it could look like
